src/encoded/static/components/filegallery.js

```diff
@@ -1134,37 +1135,89 @@ VisualizationControls.defaultProps = {
 // Displays the file filtering controls for the file association graph and file tables.
 
 class FilterControls extends React.Component {
-    constructor() {
-        super();
+    constructor(props, reactContext) {
+        super(props);
+
+        this.navigate = reactContext.navigate;
 
         // Bind this to non-React methods.
         this.handleAssemblyAnnotationChange = this.handleAssemblyAnnotationChange.bind(this);
+        this.setDownloadModalVisibility = this.setDownloadModalVisibility.bind(this);
+        this.handleDownloadClick = this.handleDownloadClick.bind(this);
+
+        this.state = { isBatchDownloadOpen: false };
+    }
+
+    setDownloadModalVisibility(isOpen) {
+        this.setState({ isBatchDownloadOpen: isOpen });
     }
 
     handleAssemblyAnnotationChange(e) {
         this.props.handleAssemblyAnnotationChange(e.target.value);
     }
 
+    handleDownloadClick() {
+        const { context, filters } = this.props;
+        const { accession } = context;
+        const assemblies = filters.assembly && filters.assembly.length > 0 && filters.assembly[0] !== 'All assemblies' ?
+            `&files.assembly=${encodedURIComponent(filters.assembly[0].replace(/\s+/g, ''))}` :
+            '';
```

Comment:
This doesn’t work though. The `filters.assembly` combines `assembly` and `genome_annotation` which you doesn’t match anything. By deleting the space with `filters.assembly[0]` === "GRCh38 V19" you get "GRCh38V19" which matches no assembly. You have to pass `files.assembly=GRCh38&files.genome_annotation=V19`. What would be great is if the individual selected assembly and genome annotation was available still split out so you can pass those two as separate query-string parameters, but if it isn't it'd be great if you could add those in somehow, in parallel with `filters`.

Reply:
Looks like `props.filterOptions` might give you the selected split assembly and annotation, as long as `filterOptions` keeps the same order as `filters`, which at a glance it looks like it does. The problem is knowing which `filterOptions` is currently selected. `props.selectedFilterValue` seems promising, but when I selected the second assembly/annotation combination, `selectedFilterValue` was "0" which doesn't help. Maybe that just needs fixing, or maybe it isn't the value I think it is.

Reply:
selectFilterValue applies to the filter by the buttons. I decided to use a different option.